### PR TITLE
Form support `labelCol` & `wrapperCol` props.

### LINKED ISSUE #15006: labelCol and wrapperCol support Inherited from Form
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
When we lay out a form, multiple form items set a repeating layout configuration. Maybe we can provide inheritable properties on the Form.

### What does the proposed API look like?
```<Form labelCol={} wrapperCol={} />```

### PR DESCRIPTION
**Pull request by zombieJ**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch.
Pull request will be merged after one of collaborators approve.
Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

close #15006

### API Realization (Optional if not new feature)

```jsx
<Form labelCol={{ span: 5 }} wrapperCol={{ span: 12 }}>
  {...}
</Form>
```

### What's the effect? (Optional if not new feature)

No effect with current usage.

### Changelog description (Optional if not new feature)

> 1. Form support `labelCol` & `wrapperCol` props.
> 2. Form 支持 `labelCol` & `wrapperCol` 属性。

### Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/form/index.zh-CN.md [hunk 1]
```diff
@@ -38,8 +38,10 @@ title: Form
 | --- | --- | --- | --- |
 | form | 经 `Form.create()` 包装过的组件会自带 `this.props.form` 属性 | object | - |
 | hideRequiredMark | 隐藏所有表单项的必选标记 | Boolean | false |
+| labelCol | label 标签布局，同 `<Col>` 组件，设置 `span` `offset` 值，如 `{span: 3, offset: 12}` 或 `sm: {span: 3, offset: 12}` | [object](https://ant.design/components/grid/#Col) |  |
 | layout | 表单布局 | 'horizontal'\|'vertical'\|'inline' | 'horizontal' |
 | onSubmit | 数据验证成功后回调事件 | Function(e:Event) |  |
+| wrapperCol | 需要为输入控件设置布局样式时，使用该属性，用法同 labelCol | [object](https://ant.design/components/grid/#Col) |  |
```
Comment (afc163) [inline components/form/index.zh-CN.md]: 写一下版本信息，防止旧版用户用了没效果。

**Review by ztplz — DISMISSED** [review]

File: components/form/index.zh-CN.md [hunk 2]
```diff
@@ -38,10 +38,10 @@ title: Form
 | --- | --- | --- | --- |
 | form | 经 `Form.create()` 包装过的组件会自带 `this.props.form` 属性 | object | - |
 | hideRequiredMark | 隐藏所有表单项的必选标记 | Boolean | false |
-| labelCol | label 标签布局，同 `<Col>` 组件，设置 `span` `offset` 值，如 `{span: 3, offset: 12}` 或 `sm: {span: 3, offset: 12}` | [object](https://ant.design/components/grid/#Col) |  |
+| labelCol | （3.14.0 新增）label 标签布局，同 `<Col>` 组件，设置 `span` `offset` 值，如 `{span: 3, offset: 12}` 或 `sm: {span: 3, offset: 12}` | [object](https://ant.design/components/grid/#Col) |  |
 | layout | 表单布局 | 'horizontal'\|'vertical'\|'inline' | 'horizontal' |
 | onSubmit | 数据验证成功后回调事件 | Function(e:Event) |  |
-| wrapperCol | 需要为输入控件设置布局样式时，使用该属性，用法同 labelCol | [object](https://ant.design/components/grid/#Col) |  |
+| wrapperCol | （3.14.0 新增）需要为输入控件设置布局样式时，使用该属性，用法同 labelCol | [object](https://ant.design/components/grid/#Col) |  |
```
Comment (afc163) [inline components/form/index.zh-CN.md]: 3.14.0 新增，之前的版本只能设置到 FormItem 上。

**Comment by zombieJ** [comment]
有个情况是 FormItem 内又套了一个 FormItem，看下来这个情况的 FormItem 没必要 pass context 的 labelCol 和 wrapperCol。

**Comment by afc163** [comment]
可以在 FormItem 的两个属性里补充读取顺序说明和版本说明。

**Comment by zombieJ** [comment]
> 可以在 FormItem 的两个属性里补充读取顺序说明和版本说明。

`读取顺序` 👌，`版本说明`我觉得没有必要。这个从 2.x 起就有了，用户照着例子写的时候就都用到了。

**Comment by afc163** [comment]
我的意思是，从 3.14.0 之后可以不用重复写这两个属性，可以从 Form 上统一读取的意思。

**Comment by zombieJ** [comment]
> 从 3.14.0 之后可以不用重复写这两个属性

Updated.

**Review by afc163 — APPROVED** [review]

**Comment by yoyo837** [comment]
这个厉害了，太厉害了，这是高手... (PS: 一条有声音的评论)

**Pull request merged**